scripts/smoke-test.js

```diff
@@ -72,14 +81,48 @@ const handlerCommand = `${path.relative(relativeBase, pythonPath)} ${path.relati
 // make custom changes to the '*.feature' files so they're able to test
 // the Python hooks (reference implementation)
 glob.sync(path.join(testDir, '**/*.feature')).forEach((featurePath) => {
-  const content = fs.readFileSync(featurePath, { encoding: 'utf-8' });
+  const content = fs.readFileSync(featurePath, 'utf8');
   const modifiedContent = uncommentPythonCodeBlocks(replacePlaceholders(content, handlerCommand));
-  fs.writeFileSync(featurePath, modifiedContent, { encoding: 'utf-8' });
+  fs.writeFileSync(featurePath, modifiedContent, 'utf8');
 });
 
 // run 'dredd-hooks-template test', should pass
 run('npx', ['dredd-hooks-template', 'test'], { cwd: testDir });
 
+
+/* ****************************************************************************
+  UPGRADE
+**************************************************************************** */
+
+// pretend 'dredd-hooks-template' is in one of the previous versions
+const projectPackagePath = path.join(testDir, 'package.json');
+const projectPackageData = fs.readJSONSync(projectPackagePath);
+projectPackageData.devDependencies['dredd-hooks-template'] = '1.0.0';
+fs.writeJSONSync(projectPackagePath, projectPackageData);
+
+// run 'dredd-hooks-template upgrade'
+const output = run('npx', ['dredd-hooks-template', 'upgrade'], {
+  cwd: testDir,
+  stdio: 'pipe',
+}).stdout.toString();
+
+// the upgrade output should contain hints
+const containsHint = output.includes('copied') && output.includes('suffixed') && output.includes('manually');
+if (!containsHint) throw new Error("Output of 'dredd-hooks-template upgrade' doesn't include hint text");
```

Comment:
Maybe semantically it would be neater to use `assert` (https://nodejs.org/api/assert.html#assert_assert_ok_value_message) (like in the tests):

```js
const assert = require('assert');
assert.ok(containsHint, "Output of 'dredd-hooks-template upgrade' doesn't include hint text");
```

TBH, it doesn't really matter though.

Reply:
I thought about it, but the rest of the file throws (each `run()` with non-zero status code will throw), so I kept it aligned. I was afraid having real assertions could confuse readers - they could think the rest of the file is just setup and these are assertions...